dags/operators/moz_dataproc_operator.py

```diff
@@ -0,0 +1,448 @@
+import os
+import re
+import uuid
+from datetime import timedelta
+
+from airflow.contrib.hooks.gcp_dataproc_hook import DataProcHook
+from airflow.exceptions import AirflowException
+from airflow.models import BaseOperator
+from airflow.utils.decorators import apply_defaults
+from airflow.utils import timezone
+from airflow.version import version
+
+"""
+We include DataprocOperationBaseOperator(unchanged) and DataprocClusterCreateOperator here to use
+the latest master branch code, which uses the v1beta2 rest api for creating clusters.
+V1beta2 allows us to install optional components e.g. Jupyter/Anaconda/Component gateway, and we
+have modified the cluster create operator to do so.
+
+"""
+class DataprocOperationBaseOperator(BaseOperator):
+    """
+    The base class for operators that poll on a Dataproc Operation.
+    """
+    @apply_defaults
+    def __init__(self,
+                 project_id,
+                 region='global',
+                 gcp_conn_id='google_cloud_default',
+                 delegate_to=None,
+                 *args,
+                 **kwargs):
+        super(DataprocOperationBaseOperator, self).__init__(*args, **kwargs)
+        self.gcp_conn_id = gcp_conn_id
+        self.delegate_to = delegate_to
+        self.project_id = project_id
+        self.region = region
+        self.hook = DataProcHook(
+            gcp_conn_id=self.gcp_conn_id,
+            delegate_to=self.delegate_to,
+            api_version='v1beta2'
```

Comment:
Note: we should make a note of this in the document we use to make note of beta apis we are leveraging.

Reply:
Done